password_security/models/res_users.py

```diff
@@ -81,6 +81,8 @@ def check_password(self, password):
             password_regex.append(r'(?=.*?\W)')
         password_regex.append('.{%d,}$' % company_id.password_length)
         if not re.search(''.join(password_regex), password):
+            self.env.cr.rollback()
```

Comment:
Wait me I don't test it with the real fail case... sorry.
I will come back with the news